src/solvers/dgsem_p4est/containers_2d.jl

```diff
@@ -37,13 +37,14 @@ end
 
 # Interpolate tree_node_coordinates to each quadrant at the specified nodes
 function calc_node_coordinates!(node_coordinates,
-                                mesh::P4estMesh{2},
-                                nodes::AbstractVector)
+                                mesh::P4estMesh{2, NDIMS_AMBIENT},
+                                nodes::AbstractVector) where {NDIMS_AMBIENT}
```

Comment:
Since this is the only place where the new functionality is used, I wonder how "complete" this feature is. Do you need to adapt other parts in TrixiAtmo.jl to make it work? Or would it work with something in Trixi.jl? Maybe it would be good to point out explicitly in the docstring that `NDIMS != NDIMS_AMBIENT` is only supported as an experimental feature for now and may not work everywhere?

Reply:
Yes, I would agree that this feature is not fully "complete" as `init_elements` is only supported with `NDIMS != NDIMS_AMBIENT` in TrixiAtmo.jl, but not Trixi.jl. For this to become a feature people can use in Trixi.jl directly, rather than a library feature used downstream, Andrés's spherical shell code (e.g. containers, contravariant vectors) would have to be moved into Trixi.jl.

Reply:
If you'd prefer not to merge this until solving PDEs on surfaces is a complete functionality that can be used directly in Trixi.jl, we are happy to work around this in TrixiAtmo.jl for now if needed, and can expand this PR into something more full featured later on.

Reply:
I'm fine having this here in Trixi.jl - but please mark it as experimental so that people don't have expectations that we cannot satisfy

Reply:
Ok, that makes sense! I have added a note in the most recent commit.